docs: what is gatsby? (#456)
vtex/faststore: docs/wip/what-is-gatsby.md

# What is Gatsby?

[GatsbyJS](https://www.gatsbyjs.com/docs/) is a [GraphQL](https://graphql.org/) and [React](https://reactjs.org/)-based static site generator that brings together the best of [Webpack](https://webpack.js.org/), React, GraphQL, and other modern tools to provide a better developer experience. 

## Why we chose Gatsby?

### No breaking changes

Most technologies used by Gatsby are already familiar to the developers working with VTEX IO. The same way as with Gatsby, our [Store Framework components](https://developers.vtex.com/vtex-developer-docs/docs/what-is-store-framework) are built with React. Bringing Gatsby and VTEX together means a fast learning curve to a universe that will allow you to build the fastest stores in the market.

### Easy data integration

Leveraging from the GraphQL power, Gatsby allows you to collect data from any source of your choosing: Markdown, JSON, third-party APIs, and even from the VTEX CMS.  
By creating an internal GraphQL server, Gatsby makes all that fetched data accessible via React components during build time. 

### Extensible architecture

> Gatsby is made more flexible and extensible thanks to its plugin-based ecosystem. Thousands of plugins can be found on the Gatsby [website] (https://www.gatsbyjs.com/plugins). When diving deeper into our new way of building stores, you'll find out more about VTEX plugins. 

Reviewer: ```suggestion
For building more complex websites, Gatsby provides thousands of plugins on its [Plugin Library](https://www.gatsbyjs.com/plugins) that lets you modularize your site customizations into specific functionalities.

Additionally, thanks to its plugin-based ecosystem, Gatsby allows you to add new functionalities to your website, making it even more flexible and extensible.
```

### High customization level

> With [Shadowing](https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/shadowing/), Gatsby increases customization levels, allowing the user to replace [webpack](https://webpack.js.org/) files with customized ones.

Reviewer: ```suggestion
With [shadowing](https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/shadowing/), Gatsby increases customization levels, allowing the developer to overwrite [webpack](https://webpack.js.org/) files with customized ones.
```

### Easy data integration

Leveraging from the GraphQL power, Gatsby allows you to collect data from any source of your choosing: Markdown, JSON, third-party APIs, and even from the VTEX CMS.  
By creating an internal GraphQL server, Gatsby makes all that fetched data accessible via React components during build time. 

### Extensible architecture

Gatsby is made more flexible and extensible thanks to its plugin-based ecosystem. Thousands of plugins can be found on the Gatsby [website] (https://www.gatsbyjs.com/plugins). When diving deeper into our new way of building stores, you'll find out more about VTEX plugins. 

### High customization level

With [Shadowing](https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/shadowing/), Gatsby increases customization levels, allowing the user to replace [webpack](https://webpack.js.org/) files with customized ones.

> A maneira como Gatsby usa GraphQL é feito para coletar seus dados de qualquer lugar: Markdown, JSON, seu CMS favorito, APIs de terceiros, em qualquer lugar! E no momento da construção, ele cria um servidor GraphQL interno de todos esses dados. Portanto, em seus componentes React seus dados são consultados em tempo de construção daquele mesmo lugar, da mesma forma por meio do GraphQL. Além da forma em como o GraphQL é utilizado, o gatsby é poderoso pelo seu ecossistema baseado em plugins, onde pode-se encontrar milhares deles em sua [página](https://www.gatsbyjs.com/plugins).

Author: ```suggestion
```

# What is Gatsby?

> GatsbyJS é um gerador de sites estático moderno baseado em React e GraphQL. Que torna a experiencia do desenvolvedor muito agradável por usar as melhores partes do React, Webpack, GraphQL e outras ferramentas.

Reviewer: ```suggestion
[GatsbyJS](https://www.gatsbyjs.com/docs/) is a [GraphQL](https://graphql.org/) and [React](https://reactjs.org/)-based static site generator that brings together the best of [Webpack](https://webpack.js.org/), React, GraphQL, and other modern tools to provide a better developer experience. 
```

> ## Por que o escolhemos?

Reviewer: ```suggestion
## Why we chose Gatsby?

### No breaking changes

Most technologies used by Gatsby are already familiar to the developers working with VTEX IO. The same way as with Gatsby, our [Store Framework components](https://developers.vtex.com/vtex-developer-docs/docs/what-is-store-framework) are built with React. Bringing Gatsby and VTEX together means a fast learning curve to a universe that will allow you to build the fastest stores in the market.

### Easy data integration

Leveraging from the GraphQL power, Gatsby allows you to collect data from any source of your choosing: Markdown, JSON, third-party APIs, and even from the VTEX CMS.  
By creating an internal GraphQL server, Gatsby makes all that fetched data accessible via React components during build time. 

### Extensible architecture

Gatsby is made more flexible and extensible thanks to its plugin-based ecosystem. Thousands of plugins can be found on the Gatsby [website] (https://www.gatsbyjs.com/plugins). When diving deeper into our new way of building stores, you'll find out more about VTEX plugins. 

### High customization level

With [Shadowing](https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/shadowing/), Gatsby increases customization levels, allowing the user to replace [webpack](https://webpack.js.org/) files with customized ones.
```

> A maneira como Gatsby usa GraphQL é feito para coletar seus dados de qualquer lugar: Markdown, JSON, seu CMS favorito, APIs de terceiros, em qualquer lugar! E no momento da construção, ele cria um servidor GraphQL interno de todos esses dados. Portanto, em seus componentes React seus dados são consultados em tempo de construção daquele mesmo lugar, da mesma forma por meio do GraphQL. Além da forma em como o GraphQL é utilizado, o gatsby é poderoso pelo seu ecossistema baseado em plugins, onde pode-se encontrar milhares deles em sua [página](https://www.gatsbyjs.com/plugins).
> 

Reviewer: ```suggestion
```

> Além de podermos nos alavancar da arquitetura de plugins, também podemos aumentar nosso nível de personalização a partir de shadowing. Este recurso nos permite substituir um arquivo do `src` incluído no pacote webpack com seu próprio arquivo.

Reviewer 2: Acho que aqui daria pra citar e botar o link da docs theme-shadowing.md da gente

> Além de podermos nos alavancar da arquitetura de plugins, também podemos aumentar nosso nível de personalização a partir de shadowing. Este recurso nos permite substituir um arquivo do `src` incluído no pacote webpack com seu próprio arquivo.

Reviewer: ## Why we chose Gatsby?

### No breaking changes

Most technologies used by Gatsby are already familiar to the developers working with VTEX IO. The same way as with Gatsby, our [Store Framework components](https://developers.vtex.com/vtex-developer-docs/docs/what-is-store-framework) are built with React. Bringing Gatsby and VTEX together means a fast learning curve to a universe that will allow you to build the fastest stores in the market.

### Easy data integration

Leveraging from the GraphQL power, Gatsby allows you to collect data from any source of your choosing: Markdown, JSON, third-party APIs, and even from the VTEX CMS.  
By creating an internal GraphQL server, Gatsby makes all that fetched data accessible via React components during build time. 

### Extensible architecture

Gatsby is made more flexible and extensible thanks to its plugin-based ecosystem. Thousands of plugins can be found on the Gatsby [website] (https://www.gatsbyjs.com/plugins). When diving deeper into our new way of building stores, you'll find out more about VTEX plugins. 

### High customization level

With [Shadowing](https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/shadowing/), Gatsby increases customization levels, allowing the user to replace [webpack](https://webpack.js.org/) files with customized ones.

> Além de podermos nos alavancar da arquitetura de plugins, também podemos aumentar nosso nível de personalização a partir de shadowing. Este recurso nos permite substituir um arquivo do `src` incluído no pacote webpack com seu próprio arquivo.

Reviewer: ```suggestion
```
